samples/TerraFX/Graphics/HelloWindow.cs

```diff
@@ -33,14 +35,32 @@ public override void Cleanup()
             base.Cleanup();
         }
 
-        public override void Initialize(Application application, TimeSpan timeout)
+        /// <summary>Initializes the GUI for this sample.</summary>
+        /// <param name="application">The hosting <see cref="Application" />.</param>
+        /// <param name="timeout">The <see cref="TimeSpan" /> after which this sample should stop running.</param>
+        public override void Initialize(Application application, TimeSpan timeout) => Initialize(application, timeout, null, null);
+
+        /// <summary>Initializes the GUI for this sample.</summary>
+        /// <param name="application">The hosting <see cref="Application" />.</param>
+        /// <param name="timeout">The <see cref="TimeSpan" /> after which this sample should stop running.</param>
+        /// <param name="windowLocation">The <see cref="Vector2" /> that defines the initial window location.</param>
+        /// <param name="windowSize">The <see cref="Vector2" /> that defines the initial window client rectangle size.</param>
+        public virtual void Initialize(Application application, TimeSpan timeout, Vector2? windowLocation, Vector2? windowSize)
         {
             ExceptionUtilities.ThrowIfNull(application, nameof(application));
 
             var windowProvider = application.GetService<WindowProvider>();
 
             _window = windowProvider.CreateWindow();
             _window.SetTitle(Name);
+            if (windowLocation.HasValue)
+            {
+                _window.Relocate(windowLocation.Value);
```

Comment:
nit: If you've already checked `HasValue` then `GetValueOrDefault()` is more efficient. It doesn't particularly matter here, but its just a good habit to call the other overload, IMO.
```suggestion
                _window.Relocate(windowLocation.GetValueOrDefault());
```